wdl/transforms/draft3/src/test/scala/wdl/draft3/transforms/linking/expression/files/FileEvaluatorSpec.scala

```diff
@@ -26,4 +26,20 @@ class FileEvaluatorSpec extends AnyFlatSpec with CromwellTimeoutSpec with Matche
     )
     evaluatedFiles shouldBeValid Set(WomSingleFile("moo.txt"))
   }
+
+  it should "find files in a string expression" in {
+    val filePiece: StringPiece = StringLiteral("moo.txt")
+    val anotherFilePiece: StringPiece = StringLiteral("boo.txt")
+    val notAFilePiece: StringPiece = NewlineEscape
+    val anotherNotAFilePiece: StringPiece = BackslashEscape
+
+    val expressionElement: ExpressionElement =
+      StringExpression(Seq(filePiece, anotherFilePiece, notAFilePiece, anotherNotAFilePiece))
+
+    val evaluatedFiles = expressionElement.predictFilesNeededToEvaluate(inputs = Map.empty,
+                                                                        ioFunctionSet = NoIoFunctionSet,
+                                                                        coerceTo = WomSingleFileType
+    )
+    evaluatedFiles shouldBeValid Set(WomSingleFile("moo.txt"), WomSingleFile("boo.txt"))
+  }
```

Comment:
Do you think it makes sense to also test the error case here, or whatever it looks like in this function when the optional variable is null?